### DISCUSSION TITLE
asInt() function is not recognized in resource's manifest

### DISCUSSION THREAD

**Discussion by jhuu32** [argoproj/argo-workflows#6819] [Q&A]
Hi All,

I tried to define a workflow whose step creates a custom resource with parameterized spec (namely size field)
This size field is defined from workflow parameter 

Here's the YAM file where the input parameter is converted into integer with asInt() function.

I got this error unexpectedly from the Argo submission UI 

 Internal Server Error: templates.main.steps[0].submit-resource templates.create-memcached.resource.manifest must be a valid yaml

What's wrong? Does complex expression evaluation work properly in resource manifest definition?
When I removed the asInt(), I got a CR validation error (expected int instead of string)

Thanks

Jacques

apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: toto
spec:
  entrypoint: main
  templates:
    - name: main
      steps:
        - - name: submit-resource
            arguments:
              parameters:
                - name: size
                  value: 5
            template: create-memcached
    - name: create-memcached
      serviceAccountName: argo
      inputs:
        parameters:
          - name: size
      resource:
        action: create
        flags:
          - '-n'
          - argo
        manifest: |
          apiVersion: cache.example.com/v1alpha1
          kind: Memcached
          metadata:
            name: memcached-sample20
          spec:
            size: **{{=asInt(inputs.parameters.size)}}**

**Comment by bygui86** [comment]
@jhuu32 any update? did you find the solution?
